tests/scripts/all.sh

```diff
@@ -1522,10 +1522,93 @@ component_test_full_no_cipher () {
 
-    msg "test: full minus CIPHER"
+    msg "test: full - CIPHER - PSA_CRYPTO_C"
     make test
 }
 
+# This is a common configurator and test function that is used in:
+# - component_test_full_no_cipher_with_crypto
+# - component_test_full_no_cipher_with_crypto_config
+# It accepts 2 input parameters:
+# - $1: boolean value which basically reflects status of MBEDTLS_PSA_CRYPTO_CONFIG
+# - $2: a text string which describes the test component
+common_test_full_no_cipher_with_crypto () {
+    USE_CRYPTO_CONFIG="$1"
+    COMPONENT_DESCRIPTION="$2"
+
+    msg "build: $COMPONENT_DESCRIPTION"
+
+    scripts/config.py full
+    scripts/config.py unset MBEDTLS_CIPHER_C
+
+    if [ "$USE_CRYPTO_CONFIG" -eq 1 ]; then
+        # Direct dependencies from PSA config
+        scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_ALG_CCM
+        scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_ALG_CCM_STAR_NO_TAG
+        scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_ALG_GCM
+        scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_ALG_CMAC
+        scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_ALG_CBC_NO_PADDING
+        scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_ALG_CBC_PKCS7
+        scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_ALG_CFB
+        scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_ALG_CTR
+        scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_ALG_ECB_NO_PADDING
+        scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_ALG_OFB
+        scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_ALG_STREAM_CIPHER
+        scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_KEY_TYPE_AES
+        scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_KEY_TYPE_DES
+        scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_KEY_TYPE_CAMELLIA
+        scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_KEY_TYPE_ARIA
+    else
+        # Don't pull in cipher via PSA mechanisms
+        scripts/config.py unset MBEDTLS_PSA_CRYPTO_CONFIG
+    fi
+    # Direct dependencies
+    scripts/config.py unset MBEDTLS_CCM_C
+    scripts/config.py unset MBEDTLS_CMAC_C
+    scripts/config.py unset MBEDTLS_GCM_C
+    scripts/config.py unset MBEDTLS_NIST_KW_C
+    scripts/config.py unset MBEDTLS_PKCS12_C
+    scripts/config.py unset MBEDTLS_PKCS5_C
+    scripts/config.py unset MBEDTLS_SSL_TLS_C
+    scripts/config.py unset MBEDTLS_SSL_TICKET_C
+    # Disable cipher modes/keys that make PSA depend on CIPHER_C.
+    # Keep CHACHA20 enabled since it does not depend on CIPHER_C.
+    scripts/config.py unset-all MBEDTLS_CIPHER_MODE
+    scripts/config.py unset MBEDTLS_AES_C
+    scripts/config.py unset MBEDTLS_DES_C
+    scripts/config.py unset MBEDTLS_ARIA_C
+    scripts/config.py unset MBEDTLS_CAMELLIA_C
+    # Dependencies on AES_C
+    scripts/config.py unset MBEDTLS_CTR_DRBG_C
+    # Disable dependencies on the AEAD algs
+    scripts/config.py unset MBEDTLS_SSL_CONTEXT_SERIALIZATION
+    # Indirect dependencies
```

Comment:
Pre-existing: I think that's a big vague. I'd rather have several paragraphs like "These depend on SSL_TLS_C", "These depend on XXX", etc.

I think if we had that in `full_no_cipher` then `USE_PSA` and LMS would have been in the paragraph "These depend on `PSA_CRYPTO_C`, making it apparent that we don't need to disable them here - unless I'm missing something.

Reply:
Actually could you also tidy the "Indirect dependencies" section in the existing `full_no_cipher` component?
- I believe most TLS stuff there no longer needs to be present as it now get auto-disabled when `SSL_TLS_C` is disabled (which is done in the "direct dependencies" section);
- For the others, I'd like to have a specific justification for each (group): "depends on PSA_CRYPTO_C", "depends on XXX", etc.

Reply:
Oh, sorry, I thought that I had just to clean the new component I added, but of course it's better to improve comments also on the original one. I will do that ASAP